### ISSUE TITLE
Free https:// for everyone using mup (i.e. support Let's Encrypt)!

### ISSUE THREAD

**Issue by Obiwarn** [arunoda/meteor-up-legacy#760]
Not sure if it makes sense to integrate in mup.

But this is great stuff:
https://letsencrypt.org/

**Comment by mgarf** [comment]
+1

**Comment by RSchwan** [comment]
+1

**Comment by MagneticMule** [comment]
+1 Great stuff indeed.

**Comment by nickpolet** [comment]
Just tried to use a .pem file generated from letsencrypt with mup and received the following error,

```
Trying to initialize SSL contexts with your certificatesError loading rsa private key
```

Is there a specific route to take when using this with mup? I managed to use it just fine with an apache server, but the setup process fails with mup? Any ideas?

**Comment by MasterJames** [comment]
Mupx uses a separate nginx as a proxy maybe that's more compatible. I think SSL is all going eliptical too?
LetsEncrypt is not official until the 9th so I might look at it next week.

**Comment by nickpolet** [comment]
The error I received above was caused by not combining the "fullchain.pem" and "privkey.pem" files. After doing this, I was able to successfully deploy the SSL cert. Letsencrypt is great.

[Follow this quick guide if your having issues.](https://nickpolet.com/blog/using-letsencrypt-with-meteor-up/P82p2SNZSSfc4mGvT)

**Comment by sakulstra** [comment]
:+1 would love to see this automated in mup(x)! Thanks for your awesome work with mup and thanks for this guide @nickpolet

**Comment by mcoenca** [comment]
They have written an article about let's encrypt and regular Mup here https://voice.kadira.io/meteor-up-let-s-encrypt-based-free-ssl-d17111f69f15#.d6fmm885p

Mupx uses nginx so I'm not sure if the article applies to it since "[Let's Encrypt] nginx support is coming soon".

**Comment by derwaldgeist** [comment]
+1

Thanks a lot for pointing to this project, it worked like a charm with mupx and nginx.

**Comment by vjrj** [comment]
Letsencrypt is in [jessie-backports](https://packages.debian.org/jessie-backports/letsencrypt). I think that meteorhacks/meteord:base is based in wheezy, but just to take into account.

**Comment by krishaamer** [comment]
Would love this!

**Comment by xvendo** [comment]
You will have a downtime in your APP with the STANDALONE version of letsencrypt. You should use webroot plugin instead! https://letsencrypt.readthedocs.io/en/latest/using.html#plugins

**Comment by derwaldgeist** [comment]
@xvendo Did you happen to make this work with nginx and mupx? According to the docs, nginx support is still experimental.

**Comment by xvendo** [comment]
Yes! You have to install nginx manually. Install the lesencrypt certificate. And when your Certificate is up and running you can use mupx without any problem.
